components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/ConsentMgtPostAuthnHandler.java

```diff
@@ -285,67 +333,126 @@ protected PostAuthnHandlerFlowStatus handlePostConsent(HttpServletRequest reques
+        return isNotBlank(consentMandatoryClaims) &&
+            !Collections.disjoint(disapprovedClaims, Arrays.asList(consentMandatoryClaims.split(CLAIM_SEPARATOR)));
+    }
 
-        List<String> approvedClaims = new ArrayList<>();
-        List<String> disapprovedClaims = new ArrayList<>();
+    private List<String> buildDisapprovedClaimList(String consentRequiredClaims, List<String> approvedClaims) {
 
-        for (Map.Entry<String, String[]> entry : requestParams.entrySet()) {
-            if (entry.getKey().startsWith(consentClaimsPrefix)) {
-                String localClaimURI = entry.getKey().substring(consentClaimsPrefix.length());
-                approvedClaims.add(localClaimURI);
-            }
-        }
+        List<String> disapprovedClaims = new ArrayList<>();
 
         if (StringUtils.isNotEmpty(consentRequiredClaims)) {
             String[] requiredClaims = consentRequiredClaims.split(CLAIM_SEPARATOR);
             List<String> consentClaims = new ArrayList<>(Arrays.asList(requiredClaims));
             consentClaims.removeAll(approvedClaims);
             disapprovedClaims = consentClaims;
         }
+        return disapprovedClaims;
+    }
 
-        if (StringUtils.isNotBlank(consentMandatoryClaims) &&
-            !Collections.disjoint(disapprovedClaims, Arrays.asList(consentMandatoryClaims.split(CLAIM_SEPARATOR)))) {
+    private List<String> buildApprovedClaimList(String consentClaimsPrefix, Map<String, String[]> requestParams) {
 
-            throw new PostAuthenticationFailedException("Consent Denied for Mandatory Attributes",
-                                                        "User denied consent to share mandatory attributes.");
+        List<String> approvedClaims = new ArrayList<>();
+
+        for (Map.Entry<String, String[]> entry : requestParams.entrySet()) {
+            if (entry.getKey().startsWith(consentClaimsPrefix)) {
+                String localClaimURI = entry.getKey().substring(consentClaimsPrefix.length());
+                approvedClaims.add(localClaimURI);
+            }
         }
+        return approvedClaims;
+    }
 
-        UserConsent userConsent = new UserConsent();
+    private String joinRequestedClaims(String consentRequestedClaims, String consentMandatoryClaims) {
 
-        userConsent.setApprovedClaims(approvedClaims);
-        userConsent.setDisapprovedClaims(disapprovedClaims);
+        return Stream.of(consentMandatoryClaims, consentRequestedClaims)
+                     .filter(StringUtils::isNotBlank)
+                     .collect(Collectors.joining(CLAIM_SEPARATOR));
+    }
 
-        return userConsent;
+    private String getConsentClaimsFromContext(AuthenticationContext context, String requestedClaimsParam) {
+
+        Object consentRequestedObj = context.getParameter(requestedClaimsParam);
+        String consentRequestedClaims = null;
+
+        if (isObjectParsableToString(consentRequestedObj)) {
```

Comment:
shall we change this to : instanceOfString()

Reply:
how about `isInstanceofString`?